base/src/store.rs

```diff
@@ -0,0 +1,62 @@
+use crate::error::BlockError;
+use core::future::Future;
+use core::pin::Pin;
+use cid::Cid;
+use std::path::Path;
+
+pub type StoreResult<T> = Pin<Box<dyn Future<Output = Result<T, BlockError>> + Send>>;
+
+/// Visibility of a block.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum Visibility {
+    /// Block is not announced on the network.
+    Private,
+    /// Block is only announced on the local network.
+    Local,
+    /// Block is announced.
+    Public,
+}
+
+/// Implementable by ipld storage backends.
+pub trait Store {
```

Comment:
Perhaps a base `ReadOnlyStore` that this one `Store: ReadOnlyStore`  for implementations who might just only want `get`?

Reply:
Not sure about the use case

Reply:
For example if i just want to query store passively and not write to i, like a monitoring service or a daily weather report, etc

Reply:
Thinking about it maybe it does make sense...